### ISSUE TITLE
Uninstall [istio installed using istioctl] via manifest generation throws errors and doesn't delete mutating and validating webhooks

### ISSUE THREAD

**Issue by sachin-ks** [istio/istio#39694]
### Bug Description

I am following the Getting Started guide: https://istio.io/latest/docs/setup/getting-started/
I just installed demo profile and tried to uninstall using the given instructions in the Getting Started guide. It throws error: `no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"` (detailed output in Steps to Reproduce).
Also, it doesn't delete the mutatingwebhookconfiguration `istio-revision-tag-default` and validatingwebhookconfiguration `istiod-default-validator`.

```
$ kubectl get mutatingwebhookconfiguration
NAME                            WEBHOOKS   AGE
istio-revision-tag-default      4          4m23s
```
```
$ kubectl get validatingwebhookconfiguration
NAME                              WEBHOOKS   AGE
istiod-default-validator          1          10m
``` 

### Version

```prose
$ istioctl version
client version: 1.14.1
control plane version: 1.14.1
data plane version: 1.14.1 (6 proxies)
$ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.24.1
Kustomize Version: v4.5.4
Server Version: v1.22.9-eks-a64ea69
```

### Steps to Reproduce
1. Install istio 
```
$ istioctl install --set profile=demo -y
✔ Istio core installed                                                                                                                                                                     
✔ Istiod installed                                                                                                                                                                         
✔ Ingress gateways installed                                                                                                                                                               
✔ Egress gateways installed                                                                                                                                                                
✔ Installation complete                                                                                                                                                                    Making this installation the default for injection and validation.

Thank you for installing Istio 1.14.  Please take a few minutes to tell us about your install/upgrade experience!  https://forms.gle/yEtCbt45FZ3VoDT5A
```
2. Uninstall Istio (notice the errors at the end of the output)
```
$ istioctl manifest generate --set profile=demo | kubectl delete --ignore-not-found=true -f -
customresourcedefinition.apiextensions.k8s.io "authorizationpolicies.security.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "destinationrules.networking.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "envoyfilters.networking.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "gateways.networking.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "istiooperators.install.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "peerauthentications.security.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "proxyconfigs.networking.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "requestauthentications.security.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "serviceentries.networking.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "sidecars.networking.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "telemetries.telemetry.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "virtualservices.networking.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "wasmplugins.extensions.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "workloadentries.networking.istio.io" deleted
customresourcedefinition.apiextensions.k8s.io "workloadgroups.networking.istio.io" deleted
serviceaccount "istio-egressgateway-service-account" deleted
serviceaccount "istio-ingressgateway-service-account" deleted
serviceaccount "istio-reader-service-account" deleted
serviceaccount "istiod" deleted
serviceaccount "istiod-service-account" deleted
clusterrole.rbac.authorization.k8s.io "istio-reader-clusterrole-istio-system" deleted
clusterrole.rbac.authorization.k8s.io "istio-reader-istio-system" deleted
clusterrole.rbac.authorization.k8s.io "istiod-clusterrole-istio-system" deleted
clusterrole.rbac.authorization.k8s.io "istiod-gateway-controller-istio-system" deleted
clusterrole.rbac.authorization.k8s.io "istiod-istio-system" deleted
clusterrolebinding.rbac.authorization.k8s.io "istio-reader-clusterrole-istio-system" deleted
clusterrolebinding.rbac.authorization.k8s.io "istio-reader-istio-system" deleted
clusterrolebinding.rbac.authorization.k8s.io "istiod-clusterrole-istio-system" deleted
clusterrolebinding.rbac.authorization.k8s.io "istiod-gateway-controller-istio-system" deleted
clusterrolebinding.rbac.authorization.k8s.io "istiod-istio-system" deleted
validatingwebhookconfiguration.admissionregistration.k8s.io "istio-validator-istio-system" deleted
configmap "istio" deleted
configmap "istio-sidecar-injector" deleted
mutatingwebhookconfiguration.admissionregistration.k8s.io "istio-sidecar-injector" deleted
deployment.apps "istio-egressgateway" deleted
deployment.apps "istio-ingressgateway" deleted
deployment.apps "istiod" deleted
Warning: policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+; use policy/v1 PodDisruptionBudget
poddisruptionbudget.policy "istio-egressgateway" deleted
poddisruptionbudget.policy "istio-ingressgateway" deleted
poddisruptionbudget.policy "istiod" deleted
role.rbac.authorization.k8s.io "istio-egressgateway-sds" deleted
role.rbac.authorization.k8s.io "istio-ingressgateway-sds" deleted
role.rbac.authorization.k8s.io "istiod" deleted
role.rbac.authorization.k8s.io "istiod-istio-system" deleted
rolebinding.rbac.authorization.k8s.io "istio-egressgateway-sds" deleted
rolebinding.rbac.authorization.k8s.io "istio-ingressgateway-sds" deleted
rolebinding.rbac.authorization.k8s.io "istiod" deleted
rolebinding.rbac.authorization.k8s.io "istiod-istio-system" deleted
service "istio-egressgateway" deleted
service "istio-ingressgateway" deleted
service "istiod" deleted
resource mapping not found for name: "stats-filter-1.11" namespace: "istio-system" from "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
ensure CRDs are installed first
resource mapping not found for name: "stats-filter-1.12" namespace: "istio-system" from "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
ensure CRDs are installed first
resource mapping not found for name: "stats-filter-1.13" namespace: "istio-system" from "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
ensure CRDs are installed first
resource mapping not found for name: "stats-filter-1.14" namespace: "istio-system" from "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
ensure CRDs are installed first
resource mapping not found for name: "stats-filter-1.15" namespace: "istio-system" from "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
ensure CRDs are installed first
resource mapping not found for name: "tcp-stats-filter-1.11" namespace: "istio-system" from "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
ensure CRDs are installed first
resource mapping not found for name: "tcp-stats-filter-1.12" namespace: "istio-system" from "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
ensure CRDs are installed first
resource mapping not found for name: "tcp-stats-filter-1.13" namespace: "istio-system" from "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
ensure CRDs are installed first
resource mapping not found for name: "tcp-stats-filter-1.14" namespace: "istio-system" from "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
ensure CRDs are installed first
resource mapping not found for name: "tcp-stats-filter-1.15" namespace: "istio-system" from "STDIN": no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3"
ensure CRDs are installed first
```
3. Check the mutatingwebhookconfiguration
```
$ kubectl get mutatingwebhookconfiguration
NAME                            WEBHOOKS   AGE
istio-revision-tag-default      4          4m23s
```
4. Check the validatingwebhookconfiguration
```
$ kubectl get validatingwebhookconfiguration
NAME                              WEBHOOKS   AGE
istiod-default-validator          1          10m
```

**Comment by sachin-ks** [comment]
@litong01 Thanks for the reply. Which version should I try to test? I used `istioctl install --set profile=demo -y` to install and `istioctl manifest generate --set profile=demo | kubectl delete --ignore-not-found=true -f -` to uninstall. I have updated the title.

**Comment by litong01** [comment]
@sachin-ks if you can try the latest master branch, that will be good. Also, I do not think you should use the install then the manifest generation to uninstall, I think that the normal process should be this, if using manifest install, then use generated manifest to uninstall. If use istioctl install, use istioctl uninstall to uninstall.  With the following command, you should be fine.

```
istioctl install --set profile=demo

istioctl x uninstall --revision=default

```

I have also tried using the generated manifest to install, then use the same manifest file to uninstall, webhooks were also removed though as you have noted, the envoy filters error showed up, that was due to the generated manifest file has envoyfilter crd at the top, the actual filters at the bottom, when crd gets deleted, the resources also got deleted as well, then when moving on to remove the filter resource, they should have been gone, so that is correct behavior.

**Comment by litong01** [comment]
generated manifest file does not have this mutatingwebhookconfiguration `istio-revision-tag-default`, only istioctl install creates that webhook. That is the reason why it was not removed when you use manifest for uninstall.

I believe that the PR I referred earlier has also been cherry picked to 1.14, if you get the 1.14 release which contains that PR, you should also be ok. after your test, if the issue is fixed, please close this issue.

**Comment by sachin-ks** [comment]
@litong01 Thanks for the reply. I used 1.14.1 for my tests when I filed the issue. Since you mentioned, the behaviour that I am seeing is correct, I am confused, what should I verify? 

If the behaviour is expected, then the Getting Started guide needs to be changed. I followed the exact same steps that are mentioned here: https://istio.io/latest/docs/setup/getting-started/

"if using manifest install, then use generated manifest to uninstall. If use istioctl install, use istioctl uninstall to uninstall."  I feel somewhere the documentation explicitly needs to say this.
